### PR TITLE
add content type header

### PR DESCRIPTION
**Pull request by bulicamihai**

### CODE REVIEW

**Comment by bulicamihai** [comment]
[AUDIO: Change MAM, notifications-push, notifications-rw, upp-content-validator, content-collection-unfolder](https://trello.com/c/iBXe9YDd/174-change-mam-notifications-push-notifications-rw-upp-content-validator-content-collection-unfolder)

File: methode-article-mapper.yaml [hunk 1]
```diff
@@ -137,6 +139,12 @@ additionalNativeContentProperties:
   lastModified: lastModified
   publishReference: publishReference
 
+kafkaContentMessage:
+  Article: application/vnd.ft-upp-article+json
+  DynamicContent: application/vnd.ft-upp-dynamic-content+json
+  ContentPackage: application/vnd.ft-upp-content-package+json
+  Audio         : application/vnd.ft-upp-audio+json
```
Comment (MMoisa) [inline methode-article-mapper.yaml:146]: Audio doesn't get processed by MAM

File: src/main/java/com/ft/methodearticlemapper/configuration/MethodeArticleMapperConfiguration.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
     private final String webUrlTemplate;
     private final String canonicalWebUrlTemplate;
     private final AppInfo appInfo;
+    private final Map<String, String> contentTypeHeader;
```
Comment (MMoisa) [inline src/main/java/com/ft/methodearticlemapper/configuration/MethodeArticleMapperConfiguration.java:38]: Rename this field to match the names used in the configuration file

File: src/main/java/com/ft/methodearticlemapper/messaging/MessageBuilder.java [hunk 3]
```diff
@@ -82,17 +85,22 @@ private Message buildMessage(String uuid, String publishReference, MessageBody m
         public final String contentUri;
         @JsonProperty("lastModified")
         public final String lastModified;
+        @JsonProperty("contentTypeHeader")
```
Comment (MMoisa) [inline src/main/java/com/ft/methodearticlemapper/messaging/MessageBuilder.java:88]: The message content type needs to be set in the `Message` `contentType` field, not in the `MessageBody`

File: src/main/java/com/ft/methodearticlemapper/messaging/MessageBuilder.java [hunk 4]
```diff
@@ -29,18 +30,20 @@
...
         this.objectMapper = objectMapper;
+        this.contentTypeHeader = contentTypeHeader;
     }
 
     Message buildMessage(Content content) {
         MessageBody msgBody = new MessageBody(
                 content,
                 contentUriBuilder.build(content.getUuid()).toString(),
-                RFC3339_FMT.format(OffsetDateTime.ofInstant(content.getLastModified().toInstant(), UTC))
+                RFC3339_FMT.format(OffsetDateTime.ofInstant(content.getLastModified().toInstant(), UTC)), contentTypeHeader
```
Comment (MMoisa) [inline src/main/java/com/ft/methodearticlemapper/messaging/MessageBuilder.java:46]: You need to set one of the values from the map as the value of the `contentType` based on the type of the content. You just inserted the whole map into the MessageBody.

**Pull request closed without merging**